source/slang/parser.cpp

```diff
@@ -347,11 +366,37 @@ namespace Slang
         }
     }
 
-    Token Parser::ReadToken()
+    TokenType Parser::PeekTokenType()
+    {
+        if (genericCloserCount)
+            return TokenType::OpGreater;
+        return tokenReader.PeekTokenType();
+    }
+
+    Token Parser::PeekToken()
+    {
+        if (genericCloserCount)
+        {
+            return Token(TokenType::OpGreater, UnownedStringSlice(">"), tokenReader.PeekToken().loc, 0);
```

Comment:
I wouldn't expect the `loc` on the token to be accurate in this case, since it is using the location of the next full token.